cmd/convert.go

```diff
@@ -0,0 +1,124 @@
+			Provider:               strings.ToLower(GlobalProvider),
+			CreateD:                ConvertDeployment,
+			CreateDS:               ConvertDaemonSet,
+			CreateRC:               ConvertReplicationController,
+			CreateDeploymentConfig: ConvertDeploymentConfig,
+			EmptyVols:              ConvertEmptyVols,
+		}
+
+		// Validate before doing anything else. Use "bundle" if passed in.
+		app.ValidateFlags(GlobalBundle, args, cmd.Flags().Lookup("file").Changed, &ConvertOpt)
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+
+		app.Convert(ConvertOpt)
+	},
+}
+
+func init() {
+
+	// Kubernetes only
+	convertCmd.Flags().BoolVarP(&ConvertChart, "chart", "c", false, "Create a Helm chart for converted objects")
+	convertCmd.Flags().BoolVar(&ConvertDaemonSet, "daemon-set", false, "Generate a Kubernetes daemonset object")
+	convertCmd.Flags().BoolVar(&ConvertReplicationController, "replication-controller", false, "Generate a Kubernetes replication controller object")
+	convertCmd.Flags().MarkHidden("chart")
+	convertCmd.Flags().MarkHidden("daemon-set")
+	convertCmd.Flags().MarkHidden("replication-controller")
+
+	// OpenShift only
+	convertCmd.Flags().BoolVarP(&ConvertDeployment, "deployment", "d", false, "Generate an OpenShift deployment object")
+	convertCmd.Flags().BoolVar(&ConvertDeploymentConfig, "deployment-config", true, "Generate an OpenShift deploymentconfig object")
+	convertCmd.Flags().MarkHidden("deployment")
+	convertCmd.Flags().MarkHidden("deployment-config")
+
+	// Standard between the two
+	convertCmd.Flags().BoolVarP(&ConvertYaml, "yaml", "y", false, "Generate resource files into yaml format")
+	convertCmd.Flags().BoolVar(&ConvertStdout, "stdout", false, "Print converted objects to stdout")
+	convertCmd.Flags().BoolVar(&ConvertEmptyVols, "emptyvols", false, "Use Empty Volumes. Do not generate PVCs")
+	convertCmd.Flags().StringVarP(&ConvertOut, "out", "o", "", "Specify a file name to save objects to")
+	convertCmd.Flags().IntVar(&ConvertReplicas, "replicas", 1, "Specify the number of repliaces in the generate resource spec")
+
+	// In order to 'separate' both OpenShift and Kubernetes only flags. A custom help page is created
+	customHelp := `Usage:{{if .Runnable}}
+  {{if .HasAvailableFlags}}{{appendIfNotPresent .UseLine "[flags]"}}{{else}}{{.UseLine}}{{end}}{{end}}{{if .HasAvailableSubCommands}}
+  {{ .CommandPath}} [command]{{end}}{{if gt .Aliases 0}}
+
+Aliases:
+  {{.NameAndAliases}}
+{{end}}{{if .HasExample}}
+
+Examples:
+{{ .Example }}{{end}}{{ if .HasAvailableSubCommands}}
+Available Commands:{{range .Commands}}{{if .IsAvailableCommand}}
+  {{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{end}}{{ if .HasAvailableLocalFlags}}
+
+Resource Flags:
+  -c, --chart                    Create a Helm chart for converted objects
+      --daemon-set               Generate a daemonset object
+  -d, --deployment               Generate a deployment object
+      --deployment-config        Generate a deployment config object
```

Comment:
@sebgoa thoughts? is there a compromise that we can achieve so we are able to distinguish each provider or does `--deployment-config` need clarification since it's not common to see it within Kubernetes?